contracts/CapabilityFilter.cdc

```diff
@@ -54,6 +54,14 @@ pub contract CapabilityFilter {
             }
         }
 
+        /// Removes all types from the mapping of denied types
+        ///
+        pub fun removeAllTypes() {
+            for type in self.deniedTypes.keys {
```

Comment:
Should we instead make `deniedTypes` a var and just set it to an empty array?

Reply:
will that be backward compatible change

Reply:
also the events might be nice to have?

Reply:
we can probably have an event that says types cleared to make it more efficient. and lemme check if it's a backward compatible change, my gut says no

Reply:
> will that be backward compatible change

I believe changing `let` to `var` is an an upgradable change yes

> also the events might be nice to have?

:+1: Sure, totally fine to add events. That might require some discussion so if you need some way to remove types right now for testing purposes, I'd split them up.

Reply:
is it worth it to do the let/var change if we want to iterate through every key to emit each event?

Reply:
Will it be helpful for each removed type to be included in its own event or could we emit an array of removed types?

Reply:
Emitting an array is what I was thinking, yeah. It depends on what kinds of events you're thinking about @rrrkren. Is there any high-level thought on that yet that we can use to think through this?

Reply:
this is what we are doing right now to reset filters:
```
        let filterDetails = filter.getDetails() as! {String:AnyStruct}
        for allowedType in filterDetails["allowedTypes"]! as! [Type]{
            filter.removeType(allowedType)
        }
```

I'd prefer to keep this PR as is if possible since it's just a sugared iterate remove all and doesn't require downstream event aggregations to change. Happy to discuss further on the single filter reset event as a follow up.